vectorstores/alloydb/README.md

```diff
@@ -0,0 +1,147 @@
+
+## Engine Creation
+
+The `AlloyDBEngine` configures a connection pool to your AlloyDB database. 
+
+```go
+package main
+
+import (
+  "context"
+  "fmt"
+
+  "github.com/tmc/langchaingo/internal/alloydbutil"
+)
+
+func NewAlloyDBEngine(ctx context.Context) (*alloydbutil.PostgresEngine, error) {
+	// Call NewPostgresEngine to initialize the database connection
+    pgEngine, err := alloydbutil.NewPostgresEngine(ctx,
+        alloydbutil.WithUser("my-user"),
+        alloydbutil.WithPassword("my-password"),
+        alloydbutil.WithDatabase("my-database"),
+        alloydbutil.WithAlloyDBInstance("my-project-id", "region", "my-cluster", "my-instance"),
+    )
+    if err != nil {
+        return nil, fmt.Errorf("Error creating PostgresEngine: %s", err)
+    }
+    return pgEngine, nil
+}
+
+func main() {
+    ctx := context.Background()
+    alloyDBEngine, err := NewAlloyDBEngine(ctx)
+    if err != nil {
+         return nil, err
+    }
+}
+```
+
+See the full [Vector Store example and tutorial](https://github.com/tmc/langchaingo/tree/main/examples/google-alloydb-vectorstore-example).
+
+## Engine Creation WithPool
+
+Create an AlloyDBEngine with the `WithPool` method to connect to an instance of AlloyDB Omni or to customize your connection pool.
+
+
+```go
+package main
+
+import (
+  "context"
+  "fmt"
+
+  "github.com/jackc/pgx/v5/pgxpool"
+  "github.com/tmc/langchaingo/internal/alloydbutil"
+)
+
+func NewAlloyDBWithPoolEngine(ctx context.Context) (*alloydbutil.PostgresEngine, error) {
+    myPool, err := pgxpool.New(ctx, os.Getenv("DATABASE_URL"))
+    if err != nil {
+        return err
+    }
+	// Call NewPostgresEngine to initialize the database connection
+    pgEngineWithPool, err := alloydbutil.NewPostgresEngine(ctx,
+        alloydbutil.WithUser("my-user"),
+        alloydbutil.WithPassword("my-password"),
+        alloydbutil.WithDatabase("my-database"),
+        alloydbutil.WithPool(myPool)
+    )
+    if err != nil {
+        return nil, fmt.Errorf("Error creating PostgresEngine with pool: %s", err)
+    }
+    return pgEngineWithPool, nil
+}
+
+func main() {
+    ctx := context.Background()
+    alloyDBEngine, err := NewAlloyDBWithPoolEngine(ctx)
+    if err != nil {
+        return nil, err
+    }
+}
+```
+
+## Vector Store Usage
+
+Use a vector store to store embedded data and perform vector search.
+
+```go
+package main
+
+import (
+  "context"
+  "fmt"
+
+  "github.com/tmc/langchaingo/embeddings"
+  "github.com/tmc/langchaingo/internal/alloydbutil"
+  "github.com/tmc/langchaingo/llms/googleai/vertex"
+  "github.com/tmc/langchaingo/vectorstores/alloydb"
+)
+
+func main() {
+    ctx := context.Background()
+    alloyDBEngine, err := NewAlloyDBEngine(ctx)
+    if err != nil {
+        return nil, err
+    }
+
+    // Initialize VertexAI LLM
+    llm, err := vertex.New(ctx, vertex.WithCloudProject("my-project-id"), vertex.WithCloudLocation("my-vertex-locations"), vertex.WithDefaultModel("text-embedding-005"))
```

Comment:
We need to verify `vertex.WithCloudProject(`. This does not run for me.

Reply:
This will be changed as soon as the PR which fixes this is done.